lib/repl.js

```diff
@@ -888,56 +904,70 @@ function REPLServer(prompt,
     self.eval(evalCmd, self.context, getREPLResourceName(), finish);
 
     function finish(e, ret) {
-      debug('finish', e, ret);
-      ReflectApply(_memory, self, [cmd]);
+      try {
+        (() => {
```

Comment:
Instead of an IIFE, you could use a `do{  }while(0);` pattern, and use `break` instead of `return`.